src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -124,11 +126,35 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
                 await lifecycleHook.AfterResourcesCreatedAsync(_model, cancellationToken).ConfigureAwait(false);
             }
         }
+        catch
+        {
+            _shutdownCts.Cancel();
+            throw;
+        }
         finally
         {
             AspireEventSource.Instance.DcpModelCreationStop();
         }
     }
+
+    public async Task StopAsync(CancellationToken cancellationToken)
+    {
+        _shutdownCts.Cancel();
+        var shutdownTasks = new List<Task>();
+        if (_resourceWatchTask is { } resourceTask)
+        {
+            shutdownTasks.Add(resourceTask);
+        }
+
+        foreach (var (cts, logTask) in _logStreams.Values)
+        {
+            cts.Cancel();
+            shutdownTasks.Add(logTask);
+        }
+
+        await Task.WhenAll(shutdownTasks).WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Do you care about exceptions?

Reply:
Exceptions are caught & logged within the individual log streamer tasks already

Reply:
Still, I'd ignore exceptions here.

Reply:
SGTM. Updated